lib/docker/imageservice.go

```diff
@@ -364,7 +460,7 @@ func newCertPool(CAFiles []string) (*x509.CertPool, error) {
 	return certPool, nil
 }
 
-func initTransport(req RegistryConnectionRequest) (http.RoundTripper, string, error) {
+func initTransport(req RegistryConnectionRequest) (http.RoundTripper, string, *multiScopeTokenHandler, error) {
```

Comment:
Frankly, I would put the result into some registryConnectResponse:
```go
type registryConnectResponse struct {
	transport    http.Transport
	registrAddr  string
	tokenHandler *multiScopeTokenHandler
}
```
or returned a partially initialized `remoteStore` - since it's what `initTransport` is for in the first place.